c10/core/TensorImpl.cpp

```diff
@@ -15,6 +15,16 @@ C10_DEFINE_int64(
     "The maximum memory in bytes to keep on shrink, if the difference between "
     "tensor sizes is bigger than this then tensor will be reset.");
 
+const std::string err_msg_tensor_metadata_change_not_allowed =
+    "is not allowed on a Tensor created from .data or .detach().\n"
+    "If your intent is to change the metadata of a Tensor without autograd tracking the metadata change,\n"
```

Comment:
can we avoid mentioning metadata here? it is an implementation detail and is not a concept that users know about...